bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,345 @@
+        String uid = name;
+        var map = channelUidMap;
+        if (name.contains("x100")) {
+            map = channelX100UidMap;
+            uid = removeX100(uid);
+        }
+        uid = uid.replaceAll("[^[\\w-]*]", "_");
+        final var firstUid = uid;
+        var idx = 1;
+        while (map.containsKey(uid)) {
+            uid = firstUid + "_" + idx++;
+        }
+        map.put(uid, name);
+        return uid;
+    }
+
+    private String buildChannelDisplayName(final String displayName) {
+        if (displayName.contains("x100")) {
+            return removeX100(displayName);
+        }
+        return displayName;
+    }
+
+    private static String removeX100(String name) {
+        var withoutSuffix = name.replace("_x100", "").replace("x100", "");
+        if (withoutSuffix.endsWith("_")) {
+            withoutSuffix = withoutSuffix.substring(0, withoutSuffix.length() - 2);
+        }
+        return withoutSuffix;
+    }
+
+    private String inOrOut(@Nullable String direction, String in, String out) {
+        if ("output".equalsIgnoreCase(direction)) {
+            return out;
+        }
+        if ("input".equalsIgnoreCase(direction)) {
+            return in;
+        }
+
+        logger.warn("Direction [{}] is unknown!", direction);
+        return out;
+    }
+
+    private void updateChannels(final List<Channel> channels) {
+        var thingBuilder = editThing();
+        thingBuilder.withChannels(channels);
+        updateThing(thingBuilder.build());
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                handleRefreshCommand(channelUID);
+            } else if (command instanceof OnOffType typedCommand) {
+                handleBoolCommand(channelUID, typedCommand == OnOffType.ON);
+            } else if (command instanceof UpDownType typedCommand) {
+                handleBoolCommand(channelUID, typedCommand == UpDownType.UP);
+            } else if (command instanceof OpenClosedType typedCommand) {
+                handleBoolCommand(channelUID, typedCommand == OpenClosedType.OPEN);
+            } else if (command instanceof PercentType typedCommand) {
+                handleDecimalCommand(channelUID, typedCommand.as(DecimalType.class));
+            } else if (command instanceof DecimalType typedCommand) {
+                handleDecimalCommand(channelUID, typedCommand);
+            } else if (command instanceof StringType typedCommand) {
+                handleStringCommand(channelUID, typedCommand);
+            } else {
+                logger.warn("Does not know how to handle command `{}` ({}) on channel `{}`!", command,
+                        command.getClass().getSimpleName(), channelUID);
+            }
+        } catch (Exception ex) {
+            logger.error("Error occurred while handling command `{}` ({}) on channel `{}`!", command,
```

Comment:
What possible Exceptions would be thrown when the commands are handeld, please use checked exceptions instead and route them to Thin status update if possible.